# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/phoc/hotfix-wlroots.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- phoc-v0.57.0/meson.build.vanilla	2026-09-26 21:54:29.720787828 +0300
+++ phoc-v0.57.0/meson.build	2026-09-26 21:55:06.543704917 +0300
@@ -53,24 +53,8 @@
 gvdb_dep = subproject('gvdb', default_options: ['tests=false']).get_variable('gvdb_dep')
 embed_wlroots = get_option('embed-wlroots')
 
-if not embed_wlroots.disabled()
-  # Try first to find wlroots as a subproject, then as a system dependency
-  wlroots_proj = subproject(
-    'wlroots',
-    default_options: ['examples=false', 'renderers=gles2,vulkan'],
-    required: false,
-  )
+wlroots = dependency('wlroots-0.20', version: '>= 0.20.2')
 
-  if embed_wlroots.enabled() and not wlroots_proj.found()
-    error('Wlroots subproject not found but enabled')
-  endif
-endif
-
-if not embed_wlroots.disabled() and wlroots_proj.found()
-  wlroots = wlroots_proj.get_variable('wlroots')
-else
-  wlroots = dependency('wlroots-0.20', version: '>= 0.20.2')
-endif
 wlroots_has_xwayland = wlroots.get_variable(pkgconfig: 'have_xwayland', internal: 'have_xwayland') == 'true'
 
 if get_option('xwayland').enabled() and not wlroots_has_xwayland
@@ -173,7 +173,6 @@
 summary(
   {
     'wlroots version': wlroots.version(),
-    'wlroots as subproject': not embed_wlroots.disabled() and wlroots_proj.found(),
     'Xwayland': have_xwayland,
     'Documentation': get_option('gtk_doc'),
     'Manual pages': get_option('man'),

